From c221ab82451f136101ff5f37604537b982042a36 Mon Sep 17 00:00:00 2001
From: Slawomir Rosek <slawomir.rosek@cavium.com>
Date: Tue, 27 Nov 2018 17:31:01 +0300
Subject: [PATCH 0767/1051] octeontx-fpa: Reset pool config on VF teardown

Reset FPA pool configuration on VF teardown.

Signed-off-by: Slawomir Rosek <slawomir.rosek@cavium.com>
Reviewed-by: Stanislaw Kardach <stanislaw.kardach@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../ethernet/cavium/octeontx-83xx/fpavf_main.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c
index 78e79f049bf3..d2d4a7841d34 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/fpavf_main.c
@@ -290,6 +290,7 @@ static int fpa_vf_setup(struct fpavf *fpa, u64 num_buffers, u32 buf_len,
 
 static int fpa_vf_teardown(struct fpavf *fpa)
 {
+	struct mbox_fpa_cfg cfg;
 	union mbox_data resp;
 	struct mbox_hdr hdr;
 	union mbox_data req;
@@ -377,6 +378,23 @@ static int fpa_vf_teardown(struct fpavf *fpa)
 		}
 	}
 
+	req.data = 0;
+	hdr.coproc = FPA_COPROC;
+	hdr.msg = FPA_CONFIGSET;
+	hdr.vfid = fpa->subdomain_id;
+
+	/* Reset pool configuration */
+	cfg.aid = 0;
+	cfg.pool_cfg = 0;
+	cfg.pool_stack_base = 0;
+	cfg.pool_stack_end = 0;
+	cfg.aura_cfg = 0;
+
+	ret = fpa->master->send_message(&hdr, &req, &resp, fpa->master_data,
+					&cfg);
+	if (ret || hdr.res_code)
+		return -EFAULT;
+
 	/* Finally free the stack */
 	dma_free_coherent(&fpa->pdev->dev, fpa->pool_size,
 			  fpa->pool_addr, fpa->pool_iova);
-- 
2.17.1

